### ISSUE TITLE
BrowserslistError: Unknown browser query `android all`

### ISSUE THREAD

**Issue by secretlifeof** [strapi/strapi#3476]
# WORKAROUND 

A possible workarround is to lock the version of the buggy pacakge in your own project package.json 
[solution](https://github.com/strapi/strapi/issues/3476#issuecomment-503065437)
Locking the culprit dependency like so:
`"caniuse-lite": "1.0.30000974"`

It might need a cache clean of your modules and your lock file for it to work.

---
**Describe the bug**
Strapi new project from strapi@beta-5 with postgreSQL. Strapi new without errors. Strapi admin ui build exits with error while using strapi develop.

**Steps**
1. npm install strapi@beta -g
2. strapi new project-title: db postgresSQL.
3. strapi develop

**Expected behavior**
No error and strapi start.

**Code snippets**
`Error: ./.cache/admin/src/components/InstallPluginPopup/styles.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/src/index.js):
BrowserslistError: Unknown browser query `android all`. Maybe you are using old Browserslist or made typo in query.
    at unknownQuery (/Users/FOLDER/strapi-test/strapi-test/node_modules/browserslist/index.js:204:10)
    at /Users/FOLDER/strapi-test/strapi-test/node_modules/browserslist/index.js:296:11
    at Array.reduce (<anonymous>)
    at resolve (/Users/FOLDER/strapi-test/strapi-test/node_modules/browserslist/index.js:237:18)
    at browserslist (/Users/FOLDER/strapi-test/strapi-test/node_modules/browserslist/index.js:361:16)
    at supportedBrowsers.some.supportedBrowser (/Users/FOLDER/strapi-test/strapi-test/node_modules/postcss-preset-env/index.js:438:105)
    at Array.some (<anonymous>)
    at stagedFeatures.filter.feature (/Users/FOLDER/strapi-test/strapi-test/node_modules/postcss-preset-env/index.js:438:80)
    at Array.filter (<anonymous>)
    at postcss$1.plugin.opts (/Users/FOLDER/strapi-test/strapi-test/node_modules/postcss-preset-env/index.js:438:44)
    at creator (/Users/FOLDER/strapi-test/strapi-test/node_modules/postcss/lib/postcss.js:133:35)
    at plugins.map.plugin (/Users/FOLDER/strapi-test/strapi-test/node_modules/precss/index.js:23:52)
    at Array.map (<anonymous>)
    at postcss.plugin.rawopts (/Users/FOLDER/strapi-test/strapi-test/node_modules/precss/index.js:23:38)
    at creator (/Users/FOLDER/strapi-test/strapi-test/node_modules/postcss/lib/postcss.js:133:35)
    at Object.<anonymous> (/Users/FOLDER/strapi-test/strapi-test/node_modules/strapi-admin/postcss.config.js:6:35)
    at runLoaders (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/NormalModule.js:302:20)
    at /Users/FOLDER/strapi-test/strapi-test/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/FOLDER/strapi-test/strapi-test/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/Users/FOLDER/strapi-test/strapi-test/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at Promise.resolve.then.then.catch (/Users/FOLDER/strapi-test/strapi-test/node_modules/postcss-loader/src/index.js:208:9)
    at compiler.run (/Users/FOLDER/strapi-test/strapi-test/node_modules/strapi-admin/index.js:140:23)
    at finalCallback (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/Compiler.js:220:39)
    at hooks.done.callAsync.err (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/Compiler.js:236:13)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:24:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/Hook.js:154:20)
    at onCompiled (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/Compiler.js:234:21)
    at hooks.afterCompile.callAsync.err (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/Compiler.js:631:15)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/Hook.js:154:20)
    at compilation.seal.err (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/Compiler.js:628:31)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/Hook.js:154:20)
    at hooks.optimizeAssets.callAsync.err (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/Compilation.js:1325:35)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/FOLDER/strapi-test/strapi-test/node_modules/tapable/lib/Hook.js:154:20)
    at hooks.optimizeChunkAssets.callAsync.err (/Users/FOLDER/strapi-test/strapi-test/node_modules/webpack/lib/Compilation.js:1316:32)`

**System**
- Node.js version: 10.16.0
- NPM version: 6.9.0
- Strapi version: 3.0.0@beta-5
- Database: PostgreSQL 11
- Operating system: OSX Mojave

How can I update the browser list version?

**Comment by sujeetbuddiga** [comment]
+1

**Comment by secretlifeof** [comment]
from package-lock.json

```
postcss-preset-env@^6.4.0:
  version "6.6.0"
  resolved "https://registry.yarnpkg.com/postcss-preset-env/-/postcss-preset-env-6.6.0.tgz#642e7d962e2bdc2e355db117c1eb63952690ed5b"
  integrity sha512-I3zAiycfqXpPIFD6HXhLfWXIewAWO8emOKz+QSsxaUZb9Dp8HbF5kUf+4Wy/AxR33o+LRoO8blEWCHth0ZsCLA==
  dependencies:
    autoprefixer "^9.4.9"
    browserslist "^4.4.2"
    caniuse-lite "^1.0.30000939"
    css-blank-pseudo "^0.1.4"
    css-has-pseudo "^0.10.0"
    css-prefers-color-scheme "^3.1.1"
    cssdb "^4.3.0"
    postcss "^7.0.14"
    postcss-attribute-case-insensitive "^4.0.1"
    postcss-color-functional-notation "^2.0.1"
    postcss-color-gray "^5.0.0"
    postcss-color-hex-alpha "^5.0.2"
    postcss-color-mod-function "^3.0.3"
    postcss-color-rebeccapurple "^4.0.1"
    postcss-custom-media "^7.0.7"
    postcss-custom-properties "^8.0.9"
    postcss-custom-selectors "^5.1.2"
    postcss-dir-pseudo-class "^5.0.0"
    postcss-double-position-gradients "^1.0.0"
    postcss-env-function "^2.0.2"
    postcss-focus-visible "^4.0.0"
    postcss-focus-within "^3.0.0"
    postcss-font-variant "^4.0.0"
    postcss-gap-properties "^2.0.0"
    postcss-image-set-function "^3.0.1"
    postcss-initial "^3.0.0"
    postcss-lab-function "^2.0.1"
    postcss-logical "^3.0.0"
    postcss-media-minmax "^4.0.0"
    postcss-nesting "^7.0.0"
    postcss-overflow-shorthand "^2.0.0"
    postcss-page-break "^2.0.0"
    postcss-place "^4.0.1"
    postcss-pseudo-class-any-link "^6.0.0"
    postcss-replace-overflow-wrap "^3.0.0"
    postcss-selector-matches "^4.0.0"
    postcss-selector-not "^4.0.0"
```

**Comment by davidpurkiss** [comment]
I've run into the same issue when running `strapi build`. I get the following output,

```
BrowserslistError: Unknown browser query `android all`
BrowserslistError: Unknown browser query `android all`
BrowserslistError: Unknown browser query `android all`
...
```

It seems to be related to a failure in a dependency. https://github.com/browserslist/browserslist/issues/382.

**Comment by shnigi** [comment]
Might be related to this: https://github.com/browserslist/browserslist/issues/382
Or this https://github.com/ben-eb/caniuse-lite/issues/27

**Comment by shnigi** [comment]
Can confirm it works by updating Browserslist to the latest version where they have some dirty hack:
https://twitter.com/Browserslist/status/1140911400304467968

Tried putting latest version to package-lock file 
"browserslist": "^4.6.3"

and works also by installing browserslist latest version as direct dependency

**Comment by FrancisVarga** [comment]
doesnt solve for me :D

**Comment by secretlifeof** [comment]
@shnigi 
**npm install -s browserslist@4.6.3** does not resolve it for me either. Now I have a continuous loop of "BrowserslistError: Unknown browser query `android all`" until ctrl-C.

**Comment by chrislarsen** [comment]
https://github.com/browserslist/browserslist/issues/382#issuecomment-503004262
pinning "caniuse-lite": "1.0.30000974" in my package.json removes the error

**Comment by shnigi** [comment]
I just deleted the browserslist dependency and everything works for me again. Build your dependencies again. It takes almost one minute for webpack to complete. The same error happens in create react app: https://github.com/facebook/create-react-app/issues/7239

**Comment by davidpurkiss** [comment]
> [browserslist/browserslist#382 (comment)](https://github.com/browserslist/browserslist/issues/382#issuecomment-503004262)
> pinning "caniuse-lite": "1.0.30000974" in my package.json removes the error

I had to update browserslist, caniuse-lite dependency in yarn.lock (package.json.lock) to get it working.

```
browserslist@^4.4.2, browserslist@^4.6.0, browserslist@^4.6.1, browserslist@^4.6.2:
  version "4.6.3"
  resolved "http://52.209.27.92:4873/browserslist/-/browserslist-4.6.3.tgz#0530cbc6ab0c1f3fc8c819c72377ba55cf647f05"
  integrity sha512-CNBqTCq22RKM8wKJNowcqihHJ4SkI8CGeK7KOR9tPboXUuS5Zk5lQgzzTbs4oxD8x+6HUshZUa2OyNI9lR93bQ==
  dependencies:
    caniuse-lite "^1.0.30000974"
    electron-to-chromium "^1.3.164"
    node-releases "^1.1.23"
``` 
and 

```
caniuse-lite@1.0.30000974:
  version "1.0.30000974"
  resolved "http://52.209.27.92:4873/caniuse-lite/-/caniuse-lite-1.0.30000974.tgz#b7afe14ee004e97ce6dc73e3f878290a12928ad8"
  integrity sha512-xc3rkNS/Zc3CmpMKuczWEdY2sZgx09BkAxfvkxlAEBTqcMHeL8QnPqhKse+5sRTi3nrw2pJwToD2WvKn1Uhvww==

caniuse-lite@^1.0.30000939, caniuse-lite@^1.0.30000971, caniuse-lite@^1.0.30000974:
  version "1.0.30000974"
  resolved "http://52.209.27.92:4873/caniuse-lite/-/caniuse-lite-1.0.30000974.tgz#b7afe14ee004e97ce6dc73e3f878290a12928ad8"
  integrity sha512-xc3rkNS/Zc3CmpMKuczWEdY2sZgx09BkAxfvkxlAEBTqcMHeL8QnPqhKse+5sRTi3nrw2pJwToD2WvKn1Uhvww==
```

**Comment by derrickmehaffy** [comment]
https://github.com/browserslist/browserslist/issues/382
https://github.com/ben-eb/caniuse-lite/issues/27
https://github.com/csstools/postcss-preset-env/pull/135/files

**Comment by derrickmehaffy** [comment]
TLDR: Not a Strapi problem, and upstream issue

**Comment by lukas-vollmer** [comment]
This fixed it for me:

`npm install caniuse-lite@1.0.30000974`

**Comment by Gribbs** [comment]
[this](https://github.com/browserslist/browserslist/issues/382#issuecomment-503398632) workaround did the trick for me

**Comment by lrjbrual** [comment]
Whatever solution I input all are the same: 

```
13:21:34 webpacker.1 | ℹ ｢wds｣: 404s will fallback to /index.html
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
13:21:35 webpacker.1 | BrowserslistError: Unknown browser query `android all`
```

**Comment by davidpurkiss** [comment]
The solution/workaround seems to vary depending on whether you're using npm or yarn. I had to fiddle around in the yarn.lock to get it working. See [my comment above](https://github.com/strapi/strapi/issues/3476#issuecomment-503065437).

**Comment by cmocha** [comment]
I was getting the error "BrowserlistError: Unknown browser query android all" during the "Building your admin UI" process after running the command:
 `strapi new cms --quickstart`
I also tried without the quickstart flag:
`strapi new cms`

I finally got it working by doing the following after wasting a couple evenings. Here is what I did. Hopefully this helps anyone else that might be struggling like me. Cheers!

Add `"caniuse-lite": "1.0.30000974"` to your package.json at the end of your dependencies.

Run `npm cache clean --force`
Run `rm -rf node_modules package-lock.json yarn.lock`
Run `npm install`
Run `yarn run build`
Run `yarn run develop`

That did it. Now I can play with Strapi and move on to the Strapi and Gatsby tutorial. Or maybe Strapi and Vue?

**Comment by davidkartuzinski** [comment]
> That did it. Now I can play with Strapi and move on to the Strapi and Gatsby tutorial. Or maybe Strapi and Vue?

I just revised the Strapi and Gatsby tutorial.  You'll get a good feel for Strapi and Gatsby by following it... https://blog.strapi.io/building-a-static-website-using-gatsby-and-strapi/

**Comment by lauriejim** [comment]
I close this issue. Now it's fixed.

**Comment by cmocha** [comment]
@davidkartuzinski 

> > That did it. Now I can play with Strapi and move on to the Strapi and Gatsby tutorial. Or maybe Strapi and Vue?
> 
> I just revised the Strapi and Gatsby tutorial. You'll get a good feel for Strapi and Gatsby by following it... https://blog.strapi.io/building-a-static-website-using-gatsby-and-strapi/

Enjoying following the tutorial David. Great stuff! Plan to try with Nuxt after I complete the Gatsby.
